### PR TITLE
Add tuoitre scraper tutorial draft

### PR DESCRIPTION
**Pull request by pdhlong**

### CODE REVIEW

**Comment by McSinyx** [comment]
Please use spaces instead of tabs for alignment.

File: man/tuoitre.rst [hunk 1]
```diff
@@ -0,0 +1,96 @@
+Scraping vaccine images from tuoitre
+====================================
+
+Site analysis
+-------------
+
+The site that we are going to work with to get the articles about vaccine from tuoitre.vn is:
+::
```
Comment (McSinyx) [inline man/tuoitre.rst]:
```suggestion
The site that we are going to work with to get the articles about vaccine from tuoitre.vn is::
```

File: man/tuoitre.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+   https://tuoitre.vn/vaccine.html
+
+Different form VnExpress, the href provided in <a> can't directly link to the articles, moreover, some <a> 
+have no href attribute. For example:
```
Comment (McSinyx) [inline man/tuoitre.rst]:
```suggestion
Different form VnExpress, the href provided in ``<a>`` can't directly link
to the articles, moreover, some ``<a>`` have no href attribute. For example:
```

File: man/tuoitre.rst [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+.. code-block:: html
+		
+   <a href="/vac-xin-covid-19-cua-astrazeneca-bi-che-bai-do-gia-mem-20201127105727613.htm" 
+      title="V&#7855;c xin COVID-19 c&#7911;a AstraZeneca b&#7883; chê bai do... giá m&#7873;m?" 
+      class="img212x132 pos-rlt" data-displayinslide="0">	
+	   
+   <a id="refresh-captcha">Lấy mã mới</a>
+
+For the images and their captions in the articles, we will try to get the 'src' - image source and 
+'alt' - image caption of the <img> tag.
```
Comment (McSinyx) [inline man/tuoitre.rst]:
```suggestion
For the images and their captions in the articles, we will try to get the ``src`` - image source and 
``alt`` - image caption of the ``<img>`` tag. 
```

File: man/tuoitre.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+download():
+^^^^^^^^^^^
+
+.. code-block:: python
+
+   async def download(img, dest, client):
+       """Save the images with theirs captions of the searched articles."""
+           caption, url = img.get('alt'), img.get('src')
+	   name, ext = splitext(basename(urlparse(url).path))
```
Comment (McSinyx) [inline man/tuoitre.rst]: This doesn't look right.

File: man/tuoitre.rst [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+       for a in links:
+           href = a.get('href')
+           if href is None: continue
+           url = 'http://tuoitre.vn' + href
+           if url.endswith('.htm') and 'vac' in url: yield url
+		
+From the <a> tags from we try to get the href attribute of each <a>. Since some <a> have no href attribute, 
+we will skip if the href returns None. To make the href become a recognized url, we add 'http://tuoitre.vn' 
+before the href. Finally, in order to get the appropriate articles related to vaccine, we only get the url end 
+with '.htm' and contains 'vac'.
```
Comment (McSinyx) [inline man/tuoitre.rst]: Same here (and a few other spots), please use double backticks for tags and codes.

**Review by McSinyx — COMMENTED** [review]
After few nits and we're good to go.

**Comment by pdhlong** [comment]
Sir please review my work

File: man/tuoitre.rst [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+.. code-block:: html
+		
+   <a href="/vac-xin-covid-19-cua-astrazeneca-bi-che-bai-do-gia-mem-20201127105727613.htm" 
+      title="V&#7855;c xin COVID-19 c&#7911;a AstraZeneca b&#7883; chê bai do... giá m&#7873;m?" 
+      class="img212x132 pos-rlt" data-displayinslide="0">	
+	   
+   <a id="refresh-captcha">Lấy mã mới</a>
+
+For the images and their captions in the articles, we will try to get the ``src`` - image source and 
+``alt`` - image caption of the ``<img>`` tag.
```
Comment (McSinyx) [inline man/tuoitre.rst]: Once you go emdash you never go back:

```suggestion
For the images and their captions in the articles, we will try to get the ``src``---image source and 
``alt``---image caption of the ``<img>`` tag. 
```

File: man/tuoitre.rst [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+.. code-block:: python
+
+   INDEX = 'https://tuoitre.vn/vaccine.html'
+	
+Then it is fetched and parsed in order to find all the available ``<a>`` tag.
+
+The scraper then focuses on 3 main functions articles(), scrape_image() and download().
+
+articles():
+^^^^^^^^^^^
```
Comment (McSinyx) [inline man/tuoitre.rst]: No colon at the end of section name please.

File: man/tuoitre.rst [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+       for a in links:
+           href = a.get('href')
+           if href is None: continue
+           url = 'http://tuoitre.vn' + href
+           if url.endswith('.htm') and 'vac' in url: yield url
+		
+From the ``<a>`` tags from we try to get the href attribute of each ``<a>``. Since some ``<a>`` have no href attribute, 
+we will skip if the href returns None. To make the href become a recognized url, we add 'http://tuoitre.vn' 
+before the href. Finally, in order to get the appropriate articles related to vaccine, we only get the url end 
+with '.htm' and contains 'vac'.
```
Comment (McSinyx) [inline man/tuoitre.rst]: Double backtick please 🥺

**Review by McSinyx — COMMENTED** [review]
Don't confuse me with [Drew DeVault](https://cmpwn.com/@sir).

**Review by McSinyx — APPROVED** [review]
Thanks!

**Comment by pdhlong** [comment]
Thank you sir I appreciate it!

**Comment by McSinyx** [comment]
Oops you didn't add this page to index.rst d-;

**Comment by pdhlong** [comment]
Sorry please help me this time sir!

**Comment by McSinyx** [comment]
Follow this up with tuoitre added to `man/index.rst`, and stop calling me sir, I'm no Bri'ish.

**Pull request merged**